### ISSUE TITLE
Flaky CI runs in PRs and main branch

### ISSUE THREAD

**Issue by snazy** [projectnessie/nessie#1055]
Recent CI runs occasionally fail with these failures.

Following outputs taken from https://github.com/projectnessie/nessie/runs/2294902098 

```
2021-04-08 08:36:03,266 [main] INFO  org.apache.spark.SparkContext - Successfully stopped SparkContext
Error:  Tests run: 12, Failures: 0, Errors: 12, Skipped: 0, Time elapsed: 1.39 s <<< FAILURE! - in org.projectnessie.deltalake.ITDeltaLog
Error:  org.projectnessie.deltalake.ITDeltaLog.checkExtendedApi  Time elapsed: 0.106 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithoutCondition2  Time elapsed: 0.135 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testAPI  Time elapsed: 0.091 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithCondition2  Time elapsed: 0.11 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.checkExtendedApiWithColumn  Time elapsed: 0.128 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithoutConditionUsingColumn  Time elapsed: 0.137 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithoutCondition  Time elapsed: 0.137 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.checkUpdateAllAndInsertAll  Time elapsed: 0.101 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.checkBasicApi  Time elapsed: 0.114 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithColumnCondition  Time elapsed: 0.095 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithConditionUsingColumn  Time elapsed: 0.106 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.

Error:  org.projectnessie.deltalake.ITDeltaLog.testWithCondition  Time elapsed: 0.108 s  <<< ERROR!
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.
```

Output from a test run:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.projectnessie.deltalake.ITDeltaLogBranches
Error:  WARNING: An illegal reflective access operation has occurred
Error:  WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/home/runner/.m2/repository/org/apache/spark/spark-unsafe_2.12/2.4.2/spark-unsafe_2.12-2.4.2.jar) to method java.nio.Bits.unaligned()
Error:  WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
Error:  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
Error:  WARNING: All illegal access operations will be denied in a future release
08:35:59,578 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [file:/home/runner/work/nessie/nessie/clients/deltalake/spark2/target/test-classes/logback-test.xml]
08:35:59,883 |-INFO in ch.qos.logback.classic.joran.action.LoggerContextListenerAction - Adding LoggerContextListener of type [ch.qos.logback.classic.jul.LevelChangePropagator] to the object stack
08:35:59,883 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@713497cd - Propagating DEBUG level on Logger[ROOT] onto the JUL framework
08:35:59,884 |-INFO in ch.qos.logback.classic.joran.action.LoggerContextListenerAction - Starting LoggerContextListener
08:35:59,884 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
08:35:59,889 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [console]
08:35:59,897 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
08:35:59,938 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.spark.sql.delta] to INFO
08:35:59,939 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@713497cd - Propagating INFO level on Logger[org.apache.spark.sql.delta] onto the JUL framework
08:35:59,939 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@713497cd - Propagating INFO level on Logger[ROOT] onto the JUL framework
08:35:59,939 |-INFO in ch.qos.logback.classic.joran.action.LevelAction - ROOT level set to INFO
08:35:59,939 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [console] to Logger[ROOT]
08:35:59,940 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
08:35:59,942 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@1add3e03 - Registering current configuration as safe fallback point
2021-04-08 08:36:00,671 [main] INFO  org.apache.spark.SparkContext - Running Spark version 2.4.2
2021-04-08 08:36:00,824 [main] WARN  o.a.hadoop.util.NativeCodeLoader - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2021-04-08 08:36:00,977 [main] INFO  org.apache.spark.SparkContext - Submitted application: c5097436-cbe2-4314-b4aa-962bbcf600a4
2021-04-08 08:36:01,078 [main] INFO  org.apache.spark.SecurityManager - Changing view acls to: runner
2021-04-08 08:36:01,080 [main] INFO  org.apache.spark.SecurityManager - Changing modify acls to: runner
2021-04-08 08:36:01,080 [main] INFO  org.apache.spark.SecurityManager - Changing view acls groups to: 
2021-04-08 08:36:01,081 [main] INFO  org.apache.spark.SecurityManager - Changing modify acls groups to: 
2021-04-08 08:36:01,082 [main] INFO  org.apache.spark.SecurityManager - SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(runner); groups with view permissions: Set(); users  with modify permissions: Set(runner); groups with modify permissions: Set()
2021-04-08 08:36:01,580 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,584 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,588 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,591 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,613 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,616 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,618 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,622 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,627 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,630 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,651 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,654 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,659 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,662 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,665 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,669 [main] WARN  org.apache.spark.util.Utils - Service 'sparkDriver' could not bind on a random free port. You may check whether configuring an appropriate binding address.
2021-04-08 08:36:01,684 [main] ERROR org.apache.spark.SparkContext - Error initializing SparkContext.
java.net.BindException: Cannot assign requested address: Service 'sparkDriver' failed after 16 retries (on a random free port)! Consider explicitly setting the appropriate binding address for the service 'sparkDriver' (for example spark.driver.bindAddress for SparkDriver) to the correct binding address.
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:455)
	at java.base/sun.nio.ch.Net.bind(Net.java:447)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at io.netty.channel.socket.nio.NioServerSocketChannel.doBind(NioServerSocketChannel.java:134)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.bind(AbstractChannel.java:550)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.bind(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeBind(AbstractChannelHandlerContext.java:506)
	at io.netty.channel.AbstractChannelHandlerContext.bind(AbstractChannelHandlerContext.java:491)
	at io.netty.channel.DefaultChannelPipeline.bind(DefaultChannelPipeline.java:973)
	at io.netty.channel.AbstractChannel.bind(AbstractChannel.java:248)
	at io.netty.bootstrap.AbstractBootstrap$2.run(AbstractBootstrap.java:356)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
2021-04-08 08:36:01,701 [main] INFO  org.apache.spark.SparkContext - Successfully stopped SparkContext
2021-04-08 08:36:01,744 [main] WARN  org.apache.spark.SparkContext - Another SparkContext is being constructed (or threw an exception in its constructor).  This may indicate an error, since only one SparkContext may be running in this JVM (see SPARK-2243). The other SparkContext was created at:
org.apache.spark.sql.SparkSession$Builder.getOrCreate(SparkSession.scala:926)
org.projectnessie.client.tests.AbstractSparkTest.create(AbstractSparkTest.java:62)
java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.base/java.lang.reflect.Method.invoke(Method.java:566)
org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
org.junit.jupiter.engine.extension.TimeoutExtension.interceptLifecycleMethod(TimeoutExtension.java:126)
org.junit.jupiter.engine.extension.TimeoutExtension.interceptBeforeEachMethod(TimeoutExtension.java:76)
org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
```

**Comment by rymurr** [comment]
Ive also seen this on recent iceberg builds

**Comment by snazy** [comment]
Does Iceberg CI run on github infra or ASF infra?

**Comment by rymurr** [comment]
GH

**Comment by snazy** [comment]
Interesting, maybe GH messed up something

**Comment by rymurr** [comment]
should be fixed with #1071
